distributed/client.py

```diff
@@ -3593,7 +3593,57 @@ def register_worker_callbacks(self, setup=None):
         setup : callable(dask_worker: Worker) -> None
             Function to register and run on all workers
         """
-        return self.sync(self._register_worker_callbacks, setup=setup)
+        return self.register_worker_plugin(_WorkerSetupPlugin(setup))
+
+    def register_worker_plugin(self, plugin=None, name=None):
+        """
+        Registers a lifecycle worker plugin for all current and future workers.
+
+        This registers a new object to handle setup and teardown handling for
+        workers in this cluster. The function will run immediately on all
+        currently connected workers. It will also be run upon connection by any
+        workers that are added in the future. Multiple setup functions can be
```

Comment:
'Multiple setup functions'? You meant multiple plugins. The setup function will be in order, but shouldn't the teardowns to be in reverse?

Reply:
Fixed.  Thanks both.